dandiapi/api/models/asset.py

```diff
@@ -125,25 +126,57 @@ def copy(cls, asset):
         return Asset(path=asset.path, blob=asset.blob, metadata=asset.metadata)
 
     @classmethod
-    def get_path(cls, path_prefix: str, qs: List[str]) -> Set:
+    def get_path(cls, dandiset_pk: str, version: str, path_prefix: str, qs: List[Asset]) -> Set:
         """
         Return the unique files/directories that directly reside under the specified path.
 
         The specified path must be a folder (must end with a slash).
         """
-        if not path_prefix:
-            path_prefix = '/'
         prefix_parts = [part for part in path_prefix.split('/') if part]
-        paths = set()
+        paths = {'folders': {}, 'items': {}}
+
+        # Declare base api url and kwargs for all asset reverse lookups
+        base_api_url = 'https://api.dandiarchive.org'
+        kwargs = {'versions__dandiset__pk': dandiset_pk, 'versions__version': version}
+
         for asset in qs:
-            path_parts = [part for part in asset['path'].split('/') if part]
+            path_parts = [part for part in asset.path.split('/') if part]
 
             # Pivot index is -1 (include all path parts) if prefix is '/'
             pivot_index = path_parts.index(prefix_parts[-1]) if len(prefix_parts) else -1
             base_path, *remainder = path_parts[pivot_index + 1 :]
-            paths.add(f'{base_path}/' if len(remainder) else base_path)
```

Comment:
BTW, in my PR I didn't want to touch any of the unrelated logic, but IMHO the above code does too much for what it needs to accomplish. If we assume that path_prefix is normalized to end with `/` (removed in this PR), code could be as simple as following IMHO easier to comprehend and likely faster (didn't do timing) code:
```python
    assert asset.path.startswith(path_prefix)  # just for paranoid me... not sure what happens in above if not fulfilled
    base_path = asset.path[len(path_prefix):]   # len is fast but could be even bound to a local var
    folder_idx = base_path.find('/')
    is_subfolder = folder_idx >= 0
    if is_subfolder:  # not sure actually if 0 can/should happen, might better just guard
        base_path = base_path[:folder_idx]
```

<details>
<summary>here is demo of it in work</summary> 

```python
class asset:
    path = '/d1/d2/123'

for path_prefix in '/', '/d1/', '/d1/d2/':
    assert asset.path.startswith(path_prefix)    
    base_path = asset.path[len(path_prefix):]   # len is fast but could be even bound to a local var
    folder_idx = base_path.find('/')
    is_subfolder = folder_idx >= 0
    if is_subfolder:  # not sure actually if 0 can/should happen, might better just guard
        base_path = base_path[:folder_idx]
    print(base_path, is_subfolder)
```
produces
```
d1 True
d2 True
123 False
```
</details>
curious ones could time-compare the two ;-)  or am I missing something above code does which "mine" doesn't?

Reply:
I timed both, and it seems they are roughly equal, but I think yours is simpler, so I've changed it in 673a7a8. Also, your assertion at the beginning is always true, as these assets are initially found by querying the databse for assets that begin with `path_prefix`, so that part is always guaranteed.

Reply:
THANK YOU for checking the timing! 
re `assert` -- IMHO the purpose of having those in the code (if ever) is to ensure that assumption (valid at the time of adding an `assert`) on outside behavior still holds in the future and that nothing broken outside for some reason invalidates it. But feel welcome to remove it.